windows/src/desktop/kmshell/install/UfrmInstallKeyboardLanguage.pas

```diff
@@ -173,78 +177,42 @@ function SystemLocalesEnumProc(lpLocaleString: PWideChar; dwFlags: DWORD; lParam
 end;
 
 procedure TfrmInstallKeyboardLanguage.AddLocale(lpLocaleString: PWideChar);
-var
-  ScriptsBuf: array[0..MAX_PATH] of char;
-  sLang: string;
-  Found: Boolean;
-  n: Integer;
 
-    procedure AddVariant(Lang: TInstLanguage);
+    procedure AddVariant(Tag: string; Lang: TInstLanguage);
```

Comment:
Are these BCP 47 utilities in common between Desktop and Developer?

Reply:
I'm not sure which utilities you are referring to?

Reply:
Procedures like `AddVariant()` which add a variant. Wouldn't both Desktop and Developer need to check for suppressed scripts and add variants?

Reply:
This particular code is Windows-specific, looking for variants that Windows supports. Developer won't use it because it works on the BCP 47 data alone.